src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.hpp

```diff
@@ -233,6 +233,16 @@ void GenericThreadStackManagerImpl_OpenThread<ImplClass>::_OnPlatformEvent(const
 #if CHIP_DETAIL_LOGGING
         LogOpenThreadStateChange(mOTInst, event->ThreadStateChange.OpenThread.Flags);
 #endif // CHIP_DETAIL_LOGGING
+
+        if (event->ThreadStateChange.OpenThread.Flags & OT_CHANGED_ACTIVE_DATASET)
+        {
+            char DebugText[CHIP_CONFIG_NETWORK_COMMISSIONING_DEBUG_TEXT_BUFFER_SIZE];
+            chip::MutableCharSpan DebugSpan(DebugText);
+            uint8_t outNetworkIndex = 0;
+            Thread::OperationalDataset dataset;
+            ReturnOnFailure(ThreadStackMgrImpl().GetThreadProvision(dataset));
+            sGenericThreadDriver.AddOrUpdateNetwork(dataset.AsByteSpan(), DebugSpan, outNetworkIndex);
```

Comment:
Note that `sGenericThreadDriver` is compiled conditionally `ifndef _NO_NETWORK_COMMISSIONING_DRIVER_` but the driver is generally a higher-level component than `ThreadStackManager`, so the latter shouldn't assume a specific implementation of the former.

Could the driver add its own event handler and handle this event internally? This would also prevent using `AddOrUpdateNetwork`, which also does the configuration backup, to update a local structure.

Reply:
Have added a event handler for the driver to update mStagingNetwork after thread panid changed.